perfkitbenchmarker/providers/ibmcloud/util.py

```diff
@@ -0,0 +1,212 @@
+SUBNETX1 = SUBNET_SUFFIX_EXTRA + '1'
+SUBNETX2 = SUBNET_SUFFIX_EXTRA + '2'
+SUBNETX3 = SUBNET_SUFFIX_EXTRA + '3'
+SUBNETX4 = SUBNET_SUFFIX_EXTRA + '4'
+SUBNETXS = [SUBNETX1, SUBNETX2, SUBNETX3, SUBNETX4]
+
+SUBNETS_EXTRA = {
+    SUBNETX1: ['10.101.20.0/24', '10.102.20.0/24', '10.103.20.0/24', '10.104.20.0/24', '10.105.20.0/24'],
+    SUBNETX2: ['10.101.30.0/24', '10.102.30.0/24', '10.103.30.0/24', '10.104.30.0/24', '10.105.30.0/24'],
+    SUBNETX3: ['10.101.40.0/24', '10.102.40.0/24', '10.103.40.0/24', '10.104.40.0/24', '10.105.40.0/24'],
+    SUBNETX4: ['10.101.50.0/24', '10.102.50.0/24', '10.103.50.0/24', '10.104.50.0/24', '10.105.50.0/24']
+}
+
+SUBNETS_EXTRA_GATEWAY = {
+    SUBNETX1: ['10.101.20.1', '10.102.20.1', '10.103.20.1', '10.104.20.1', '10.105.20.1'],
+    SUBNETX2: ['10.101.30.1', '10.102.30.1', '10.103.30.1', '10.104.30.1', '10.105.30.1'],
+    SUBNETX3: ['10.101.40.1', '10.102.40.1', '10.103.40.1', '10.104.40.1', '10.105.40.1'],
+    SUBNETX4: ['10.101.50.1', '10.102.50.1', '10.103.50.1', '10.104.50.1', '10.105.50.1']
+}
+
+
+def ReadConfig(config):
+  """Reads in config yml
+
+  Args:
+    config: config file name.
+
+  Returns:
+    The json representation of the config file.
+  """
+  try:
+    with open(config, 'r') as stream:
+      data = json.dumps(yaml.load(stream), sort_keys=True)
+      return json.loads(data)
+  except Exception as ex:
+    raise errors.Error('Failed to load configuration file %s, %s', config, ex)
+
+
+def GetSubnetIndex(ipv4_cidr_block):
+  """Finds the index for the given cidr
+
+  Args:
+    ipv4_cidr_block: cidr to find.
+
+  Returns:
+    The index number of the found cidr as in the predefined list
+  """
+  for name in SUBNETXS:
+    ip_list = SUBNETS_EXTRA[name]
+    for i in range(len(ip_list)):
+      if ip_list[i] == ipv4_cidr_block:
+        return i
+  return -1
+
+
+def GetRouteCommands(data, index, target_index):
+  """Creates a list of ip route commands in text format to run on vm,
+    not used on normal perfkit runs.
+
+  Args:
+    data: output from route command on vm.
+    index: subnet index.
+    target_index: target subnet index.
+
+  Returns:
+    The index number of the found cidr as in the predefined list
+  """
+  route_cmds = []
+  if data:
+    for subnet_name in SUBNETXS:
+      subnet_cidr_block = SUBNETS_EXTRA[subnet_name][index]
+      target_cidr_block = SUBNETS_EXTRA[subnet_name][target_index]
+      subnet_gateway = SUBNETS_EXTRA_GATEWAY[subnet_name][index]
+      route_entry = subnet_cidr_block.split('/24')
+      interface = None
+      for line in data.splitlines():
+        items = line.split()
+        if len(items) > 6 and items[0] == route_entry[0]:
+          interface = items[7]
+          route_cmds.append('ip route add ' + target_cidr_block + ' via ' + \
+                            subnet_gateway + ' dev ' + interface)
+  return route_cmds
+
+
+def GetMinMax(data):
+  """Finds min and max values in the given list of values"""
+  if not data:
```

Comment:
data = [float(x) for x in data]
return (min(data), max(data))

Reply:
yes, actually this is not really needed so I just removed this function.